### PR TITLE
Oss 50 create ci cd file for githib actions

### PR DESCRIPTION
**Pull request by WillReed1998**
Clean up tests and controllers 
Add simple CI/CD for tests

### CODE REVIEW

File: .github/workflows/ci-cd.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: Java CI
+
+on:
+  push:
+    branches:
+      - dev
+      - master
+      - OSS-50-Create-CI-CD-file-for-Githib-Actions
```
Comment (obaibula) [inline .github/workflows/ci-cd.yml:8]: This branch is redundant, please remove it.

File: .github/workflows/ci-cd.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  pull_request:
+    branches:
+      - dev
+      - master
+      - OSS-50-Create-CI-CD-file-for-Githib-Actions
```
Comment (obaibula) [inline .github/workflows/ci-cd.yml:13]: Same here

File: .github/workflows/ci-cd.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+name: Java CI
```
Comment (obaibula) [inline .github/workflows/ci-cd.yml:1]: Rename to something meaningful for example: Test and Build

File: .github/workflows/ci-cd.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up JDK 21
+        uses: actions/setup-java@v4
+        with:
+          java-version: '21'
+          distribution: 'adopt'
```
Comment (obaibula) [inline .github/workflows/ci-cd.yml:27]: We don't use the "adopt" distribution. In Dokerfile you use Oracle, so change it to "oracle"

File: .github/workflows/ci-cd.yml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+      - name: Build and test with Maven
+        run: mvn clean test
```
Comment (obaibula) [inline .github/workflows/ci-cd.yml:30]: To build the application you should use "package" Maven stage, so you can write something like this:


      
image

**Review by obaibula — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-cd.yml (hunk 2 above)
Comment (WillReed1998) [reply .github/workflows/ci-cd.yml:13]: For test purposes

**Comment by WillReed1998** [comment]
I assume it is fixed now

**Review by obaibula — APPROVED** [review]

**Pull request merged**